lib/hl7/message.ex

```diff
@@ -529,6 +526,30 @@ defmodule HL7.Message do
     end
   end
 
+  def maybe_transcode(binary) do
+    case String.valid?(binary) do
```

Comment:
`String.valid?/1` seems to be slightly faster than `:unicode.characters_to_binary/1`,  it may be it's still best to stick with the unicode module for this.

Reply:
I also don't think we should even check to see if the String is valid in the case accept_latin1 is true. We avoid iterating twice and if's invalid, we'll have to do the iteration anyway.